contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_hexadecimal.py

```diff
@@ -0,0 +1,161 @@
+"""
+This is a template for creating custom ColumnMapExpectations.
+For detailed instructions on how to use it, please see:
+    https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations
+"""
+
+from typing import Optional
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import PandasExecutionEngine
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesToBeHexadecimal(ColumnMapMetricProvider):
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.is_hexadecimal"
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+        def is_hex(x):
+            try:
```

Comment:
@andrewsx here you could try `isinstance(x, str)` to check whether the variable is a string. You could also have a check for `if not x:` to cover the False and None cases.

Reply:
@rdodev,  I tried the following test for None, but it does not pass. Why is that?

```
def _pandas(cls, column, **kwargs):
        def is_hex(x):
            if not x:
                return False
            if not isinstance(x, str):
                return False
            try:
                int(x, 16)
                return True
            except ValueError:
                return False

        return column.apply(is_hex)
```

"g": ["a8", "41", "ca", None, "4AA", "31"]
{
                    "title": "negative_test_with_none",
                    "include_in_gallery": True,
                    "exact_match_out": False,
                    "in": {"column": "g"},
                    "out": {
                        "success": False,
                        "unexpected_index_list": [3],
                        "unexpected_list": [None],
                    },
                }

Reply:
This is strange. Let me tinker for a bit.

Reply:
@andrewsx Can you try adding a clause `if x is None:` to make the check explicit?

Reply:
@rdodev,  I tried adding a print statement inside that block and it never prints out

Reply:
@rdodev  It seems to skip over None values (there are supposed to be 6 values in each test):
Column 0    a8
1    41
2    ca
4     0
5    31

However, when the input is an empty string, at least it seems there is a row for it.
Column 0    a8
1    41
2    ca
3      
4     0
5    31

I'm thinking the filter could be inside the decorator @column_condition_partial

Reply:
@andrewsx that makes more sense now. Let me take a look and will get back to you.

Reply:
@andrewsx are you printing before or after the logic?